src/rv/comm/rcssserver/GameState.java

```diff
@@ -335,6 +372,29 @@ public void parse(SExp exp, WorldModel world) {
                     scoreRight = Integer.parseInt(atoms[1]);
                     playStateChanges++;
                     break;
+                case FOUL:
+                    Foul foul = new Foul();
+                    foul.time = time;
+                    foul.index = Integer.parseInt(atoms[1]);
+                    foul.type = GameState.FoulType.values()[Integer.parseInt(atoms[2])];
+                    foul.team = Integer.parseInt(atoms[3]);
+                    foul.unum = Integer.parseInt(atoms[4]);
+                    foul.receivedTime = System.currentTimeMillis();
+                    boolean fAlreadyHaveFoul = false;
+                    for (int i = 0; i < fouls.size(); ++i) {
+                        Foul currentFoul = fouls.get(i);
+                        if (currentFoul.type == foul.type && currentFoul.team == foul.team
```

Comment:
I don't really know what index is.  That code was copied verbatim from Sander's implementation.  I'd have to look at the server code to see what index actually is.